### ISSUE TITLE
prop-types does mark nested destructured props as being used

### ISSUE THREAD

**Issue by kolodny** [yannickcr/eslint-plugin-react#296]
Assume I have a propTypes as follows:

``` js
propTypes: {
  foo: PropTypes.number,
  bar: PropTypes.shape({
    faz: PropTypes.number,
    qaz: PropTypes.object,
  }),
};
```

I would want accessing `this.props.bar.faz` to not generate an error, while `this.props.bar.nope` to generate an error.

This may be tricky because `PropTypes.shape` is technically a black box, but when used with something like `recursive` from https://github.com/kolodny/zan it would make sense

**Comment by yannickcr** [comment]
Nested prop-types are supported since `v2.6.0`, your example will correctly generate an error for `bar.nope` but not for `bar.faz`.

**Comment by kolodny** [comment]
Ahh so it would seem, I was doing `shape()` and not `PropTypes.shape()`, is there a way to support just `.shape()`?

**Comment by yannickcr** [comment]
Not for now, but I can consider this possibility.

**Comment by ianmstew** [comment]
In version `3.11.2` I can confirm dot-accessed nested props are checked, but destructured nested props are not.  Should I open a new issue on this?

For example, `const nope = this.props.bar.nope` will generate an error, but `const {bar:{nope}} = this.props` will not.

**Comment by lencioni** [comment]
I have also confirmed that dot-accessed nested props are checked but nested destructured props are not. Here's a repro case:

```jsx
 function Foo(props) {
   const { bar: { nope } } = props;
   return <div test={nope} />;
 }

 Foo.propTypes = {
   foo: PropTypes.number,
   bar: PropTypes.shape({
     faz: PropTypes.number,
     qaz: PropTypes.object,
   }),
 };
```

@ianmstew don't worry about opening a new issue, I'll just modify this one.
